Gems/Atom/Tools/MaterialEditor/Code/Source/Viewport/MaterialViewportWidget.cpp

```diff
@@ -74,44 +74,46 @@ namespace MaterialEditor
     {
         m_ui->setupUi(this);
 
-        // The viewport context created by AtomToolsFramework::RenderViewportWidget has no name.
-        // Systems like frame capturing and post FX expect there to be a context with DefaultViewportContextName
-        auto viewportContextManager = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get();
-        const AZ::Name defaultContextName = viewportContextManager->GetDefaultViewportContextName();
-        viewportContextManager->RenameViewportContext(GetViewportContext(), defaultContextName);
+        // Create a custom entity context for the entities in this viewport 
+        m_entityContext = AZStd::make_unique<AzFramework::EntityContext>();
+        m_entityContext->InitContext();
 
         // Create and register a scene with all available feature processors
         AZ::RPI::SceneDescriptor sceneDesc;
-        sceneDesc.m_nameId = AZ::Name("MaterialViewport");
+        sceneDesc.m_nameId = AZ::Name(AZStd::string::format("MaterialViewportWidgetScene_%i", GetViewportContext()->GetId()));
```

Comment:
`AZStd::string::format("MaterialViewportWidgetScene_%i", GetViewportContext()->GetId()` is used in a few places. Maybe it could put into a variable or even a member if it's necessary in other places.